src/service/url-replacements-impl.js

```diff
@@ -146,6 +152,22 @@ export class GlobalVariableSource extends VariableSource {
       return Services.viewerForDoc(this.ampdoc).getReferrerUrl();
     }));
 
+    // Like DOCUMENT_REFERRER, but returns null if the referrer is of
+    // same domain or the corresponding CDN proxy.
+    this.setAsync('EXTERNAL_REFERRER', /** @type {AsyncResolverDef} */(() => {
+      return Services.viewerForDoc(this.ampdoc).getReferrerUrl()
+          .then(referrer => {
+            if (!referrer) {
+              return referrer;
+            }
+            const referrerHostname = parseUrl(getSourceUrl(referrer)).hostname;
```

Comment:
is the domain-prefix on cdn.ampproject.org always going to be same as `sourceUrl.replace(/\./, '-')`?